orthopairs/src/main/java/org/reactome/release/orthopairs/Main.java

```diff
@@ -0,0 +1,155 @@
+     *
+     *  These files are much smaller than the older version, since PANTHER annotates which homologs are the 'Least Diverged', meaning we can filter out many that add noise to the dataset.
+     *
+     */
+
+    private static final Logger logger = LogManager.getLogger();
+
+    public static void main( String[] args ) throws IOException, ParseException {
+
+        String pathToConfig = "";
+        String sourceMappingSpecies = "";
+        // If using an alternative source species, specify the 4-letter code as the second argument
+        //TODO: Better solution for this. What if user wants default config but to change source species?
+        if (args.length > 0) {
+            pathToConfig = args[0];
+            sourceMappingSpecies = args[1];
+        } else {
+            pathToConfig = "src/main/resources/config.properties";
+            sourceMappingSpecies = "hsap";
+        }
+        Properties props = new Properties();
+        props.load(new FileInputStream(pathToConfig));
+        // Load all config properties
+        String releaseNumber = props.get("releaseNumber").toString();
+        String pathToSpeciesConfig = props.get("pathToSpeciesConfig").toString();
+        String pantherFilepath = props.get("pantherFileURL").toString();
+        String pantherQfOFilename = props.get("pantherQfOFilename").toString();
+        String pantherHCOPFilename = props.get("pantherHCOPFilename").toString();
+        String MGIFileURL = props.get("MGIFileURL").toString();
+        String RGDFileURL = props.get("RGDFileURL").toString();
+        String XenbaseFileURL = props.get("XenbaseFileURL").toString();
+        String ZFINFileURL = props.get("ZFINFileURL").toString();
+
+        new File(releaseNumber).mkdir();
+
+        logger.info("Starting Orthopairs file generation");
+        // Download and extract homology files from Panther
+        List<String> pantherFiles = new ArrayList<String>(Arrays.asList(pantherQfOFilename, pantherHCOPFilename));
+        for (String pantherFilename : pantherFiles) {
+            downloadAndExtractTarFile(pantherFilename, pantherFilepath);
+        }
+
+        // Download ID files from various model organism databases (Mouse Genome Informatics, Rat Genome Database, Xenbase (frog), ZFIN (Zebrafish))
+        // HGNC identifier file is downloaded as well.
+        List<String> alternativeIdMappingURLs = new ArrayList<>(Arrays.asList(MGIFileURL,RGDFileURL,XenbaseFileURL,ZFINFileURL));
+        for (String altIdURL : alternativeIdMappingURLs) {
+            File altIdFilepath = new File(altIdURL.substring(altIdURL.lastIndexOf("/")+1));
+            if (!altIdFilepath.exists()) {
+                logger.info("Downloading " + altIdURL);
+                FileUtils.copyURLToFile(new URL(altIdURL), altIdFilepath);
+            } else {
+                logger.info(altIdFilepath + " already exists");
+            }
```

Comment:
The code to download files above is used in two places, so it could be extracted to a method.

```
private static void downloadFile(URL resourceToDownload, File downloadTargetFile) throws IOException {
	// Download files
	if (!downloadTargetFile.exists()) {
		logger.info("Downloading " + resourceToDownload);
		FileUtils.copyURLToFile(resourceToDownload, downloadTargetFile);
	} else {
		logger.info(downloadTargetFile + " already exists");
	}
}
```